addons/medical_engine/functions/fnc_handleDamage.sqf

```diff
@@ -0,0 +1,105 @@
+/*
+ * Author: commy2
+ * Main HandleDamage EH function for soldiers.
+ *
+ * Arguments:
+ * Handle damage EH
+ *
+ * Return Value:
+ * Damage to be inflicted <NUMBER>
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+params ["_unit", "_selection", "_damage", "_shooter", "_ammo", "_hitPointIndex"];
+//diag_log _this;
+
+// HD sometimes triggers for remote units - ignore.
+if (!local _unit) exitWith {nil};
+
+// Get missing meta info
+private ["_hitPoint", "_oldDamage"];
+
+if (_hitPointIndex < 0) then {
+    _hitPoint = "#structural";
+    _oldDamage = damage _unit;
+} else {
+    _hitPoint = toLower (getAllHitPointsDamage _unit select 0 select _hitPointIndex);
+    _oldDamage = _unit getHitIndex _hitPointIndex;
+};
+
+private _newDamage = _damage - _oldDamage;
+_unit setVariable [format [QGVAR(newDamage$%1), _hitPoint], _newDamage];
+
+// Never kill the unit by the critical hit points of the engine.
+if (_hitPoint in ["#structural", "hitbody", "hithead"]) then {
+    _damage = _damage min 0.99;
+};
+
+// These control blood material visuals.
+// If damage is in dummy hitpoints, "hands" and "legs", don't change anything
+if (_hitPoint in ["hithead", "hitbody", "hithands", "hitlegs"]) exitWith {_oldDamage};
+
+// Add injury
+if (_hitPoint isEqualTo "ace_hdbracket") exitWith {
+    // --- Head
+    private _damageFace = _unit getVariable [QGVAR(newDamage$HitFace), 0];
+    private _damageNeck = _unit getVariable [QGVAR(newDamage$HitNeck), 0];
+    private _damageHead = (_unit getVariable [QGVAR(newDamage$HitHead), 0]) max _damageFace max _damageNeck;
+
+    // --- Body
+    private _damageStructural = _unit getVariable [QGVAR(newDamage$#structural), 0];
```

Comment:
I wonder; should this "engine" module know about the different types of damage the "body" can have. Shouldn't that just be handled by the higher-level component? 

Maybe it should just be
```sqf
private _damageBody = (_unit getVariable [QGVAR(newDamage$HitBody), 0]);
```